src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -854,90 +595,58 @@ def _runtool(self, main, tool_name, args, distribution,
           wu, self._nailgunnable_combined_classpath, main, tool_name, args, distribution),
       })()
 
-  def _run_metai_tool(self,
-                      distribution,
-                      metai_classpath,
-                      rsc_index_dir,
-                      tgt,
-                      extra_input_files=()):
-    # TODO have metai write to a different spot than metacp
-    # Currently, the metai step depends on the fact that materializing
-    # ignores existing files. It should write the files to a different
-    # location, either by providing inputs from a different location,
-    # or invoking a script that does the copying
-    args = [
-      '--verbose',
-      os.pathsep.join(metai_classpath)
-    ]
-    self._runtool(
-      'scala.meta.cli.Metai',
-      'metai',
-      args,
-      distribution,
-      tgt=tgt,
-      input_files=tuple(metai_classpath) + tuple(extra_input_files),
-      output_dir=rsc_index_dir
-    )
+  _JDK_LIB_NAMES = ['rt.jar', 'dt.jar', 'jce.jar', 'tools.jar']
 
-  def _collect_metai_classpath(self, metacp_result, relative_input_paths):
-    metai_classpath = []
-
-    relative_workdir = fast_relpath(
-      self.context.options.for_global_scope().pants_workdir,
-      get_buildroot())
-    # NB The json uses absolute paths pointing into either the buildroot or
-    #    the temp directory of the hermetic build. This relativizes the keys.
-    #    TODO remove this after https://github.com/scalameta/scalameta/issues/1791 is released
-    desandboxify = _create_desandboxify_fn(
-      [
-        os.path.join(relative_workdir, 'resolve', 'coursier', '[^/]*', 'cache', '.*'),
-        os.path.join(relative_workdir, 'resolve', 'ivy', '[^/]*', 'ivy', 'jars', '.*'),
-        os.path.join(relative_workdir, 'compile', 'rsc', '.*'),
-        os.path.join(relative_workdir, r'\.jdk', '.*'),
-        os.path.join(r'\.jdk', '.*'),
-      ]
-      )
+  @memoized_method
+  def _jdk_libs_paths_and_digest(self, hermetic_dist):
+    jdk_libs_rel, jdk_libs_globs = hermetic_dist.find_libs_path_globs(self._JDK_LIB_NAMES)
+    jdk_libs_digest = self.context._scheduler.capture_snapshots(
+      (jdk_libs_globs,))[0].directory_digest
+    return (jdk_libs_rel, jdk_libs_digest)
+
+  @memoized_method
+  def _jdk_libs_abs(self, nonhermetic_dist):
+    return nonhermetic_dist.find_libs(self._JDK_LIB_NAMES)
 
-    status_elements = {
-      desandboxify(k): desandboxify(v)
-      for k,v in metacp_result["status"].items()
-    }
+  class _HermeticDistribution(object):
```

Comment:
This could be much better documented (sorry!), and I didn't do that because I don't at all understand what the difference is between a hermetic and non-hermetic `Distribution`. I *really* think the process of obtaining the appropriate distributions (basically almost everything related to distributions in this file) should be wrapped up in `jvm_compile.py` (and this would happen to make it significantly easier to introduce it to other tools and execution strategies such as scalac and graal native-image). I will make a ticket to do this -- it might make this PR easier to review if I can do that in the meantime?

Reply:
Hm. I mean, I could revert that change. I don't think it's relevant to this changeset. Then, we could have a separate PR that does just that.